tests/unit/neuralnets/NeuralRectifiedLinearLayer_unittest.cc

```diff
@@ -31,172 +31,119 @@
  */
 
-#include <shogun/neuralnets/NeuralRectifiedLinearLayer.h>
-#include <shogun/neuralnets/NeuralInputLayer.h>
-#include <shogun/lib/SGVector.h>
+#include "NeuralLayerTestFixture.h"
 #include <shogun/lib/SGMatrix.h>
-#include <shogun/mathematics/Math.h>
-#include <gtest/gtest.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/mathematics/linalg/LinalgSpecialPurposes.h>
+#include <shogun/neuralnets/NeuralInputLayer.h>
+#include <shogun/neuralnets/NeuralRectifiedLinearLayer.h>
+
+#include <memory>
 
 using namespace shogun;
 
+using NeuralRectifiedLinearLayerTest = NeuralLayerTestFixture;
+
 /** Compares the activations computed using the layer against manually computed
  * activations
  */
-TEST(NeuralRectifiedLinearLayer, compute_activations)
+TEST_F(NeuralRectifiedLinearLayerTest, compute_activations)
 {
-	CNeuralRectifiedLinearLayer layer(9);
-
 	// initialize some random inputs
-	CMath::init_random(100);
-	SGMatrix<float64_t> x(12,3);
-	for (int32_t i=0; i<x.num_rows*x.num_cols; i++)
-		x[i] = CMath::random(-10.0,10.0);
-
-	CNeuralInputLayer* input = new CNeuralInputLayer (x.num_rows);
-	input->set_batch_size(x.num_cols);
-
-	CDynamicObjectArray* layers = new CDynamicObjectArray();
-	layers->append_element(input);
+	SGMatrix<float64_t> x;
+	CNeuralInputLayer* input;
+	std::tie(x, input) = setup_input_layer<float64_t>(12, 3, -10.0, 10.0);
 
+	// initialize the rectified linear layer
+	CNeuralRectifiedLinearLayer layer(9);
 	SGVector<int32_t> input_indices(1);
 	input_indices[0] = 0;
-
-	// initialize the layer
-	layer.initialize_neural_layer(layers, input_indices);
-	SGVector<float64_t> params(layer.get_num_parameters());
-	SGVector<bool> param_regularizable(layer.get_num_parameters());
-	layer.initialize_parameters(params, param_regularizable, 1.0);
-	layer.set_batch_size(x.num_cols);
-
-	// compute the layer's activations
-	input->compute_activations(x);
-	layer.compute_activations(params, layers);
+	auto params =
+	    init_linear_layer(&layer, input_indices, x.num_cols, 1.0, false);
 	SGMatrix<float64_t> A = layer.get_activations();
 
-	// manually compute the layer's activations
+	// Manually compute Recitified linear activations
+	auto biases =
+	    SGVector<float64_t>(params.vector, layer.get_num_neurons(), 0);
+	auto weights = SGMatrix<float64_t>(
+	    params.vector, layer.get_num_neurons(), x.num_rows,
+	    layer.get_num_neurons());
 	SGMatrix<float64_t> A_ref(layer.get_num_neurons(), x.num_cols);
-
-	float64_t* biases = params.vector;
-	float64_t* weights = biases + layer.get_num_neurons();
-
-	for (int32_t i=0; i<A_ref.num_rows; i++)
-	{
-		for (int32_t j=0; j<A_ref.num_cols; j++)
-		{
-			A_ref(i,j) = biases[i];
-
-			for (int32_t k=0; k<x.num_rows; k++)
-				A_ref(i,j) += weights[i+k*A_ref.num_rows]*x(k,j);
-
-			A_ref(i,j) = CMath::max<float64_t>(0, A_ref(i,j));
-		}
-	}
+	shogun::linalg::add_vector(
+	    shogun::linalg::matrix_prod(weights, x), biases, A_ref);
```

Comment:
you could store this in a variable as it is not optimized out or anything anyways

Reply:
Not sure what you mean? `add_vector` itself stores its result in the position denoted by A_ref and returns void.